src/driver/swim.c

```diff
@@ -318,26 +317,34 @@ int swim_entry() {
   // 128 cycles (@8MHz) i.e. 16us. We currently just use it as a sanity
   // check to make sure the (right) device responds.
   timeout = MICROS_TO_CYCLES(30) / 6;
-  while (!READ_PIN() && --timeout)
+  while (!READ_PIN(SWIM) && --timeout)
     ;  // ~6 cycles
   int duration = get_ccount() - counter;
   if (!timeout) return SWIM_ERROR_SYNC_TIMEOUT_2;
 
   esp8266_leave_critical(state);
-  sync_ccount(counter + duration +
-              24);  // Need to wait at least 300ns (=24 cycles)
+
+  SET_PIN_HIGH(SWIM);
+  PIN_AS_INPUT(SWIM);
+  sync_ccount(counter + duration + 24 ); // Need to wait at least 300ns (=24 cycles)
+
+#if(FIRMWARE_VERSION_MAJOR + FIRMWARE_VERSION_MINOR == 0)
+  reset(false);
+  sync_ccount(get_ccount() + MICROS_TO_CYCLES(1000));
```

Comment:
I assume the purpose here is to sleep 1ms. Can you document why? I think such a long sleep is better handled on the stm8flash side (I’m slightly afraid of the watchdog timer here), in particular I don’t think you need to use the finegrained instruction timer for this.

Reply:
I was refering to:
https://www.st.com/content/ccc/resource/technical/document/user_manual/ca/89/41/4e/72/31/49/f4/CD00173911.pdf/files/CD00173911.pdf/jcr:content/translations/en.CD00173911.pdf

In step 7 it says:
> Release the reset which starts the option byte loading sequence. Wait 1 ms for stabilization.

I have removed it now, since it seems to work fine and should probably better be done in stm8flash